flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -1083,8 +1136,14 @@ public void alterPartitionColumnStatistics(ObjectPath tablePath, CatalogPartitio
 	}
 
 	@Override
-	public CatalogTableStatistics getTableStatistics(ObjectPath tablePath) throws TableNotExistException, CatalogException {
-		throw new UnsupportedOperationException();
+	public CatalogTableStatistics getTableStatistics(ObjectPath tablePath) throws TableNotExistException,
+			CatalogException, TableNotPartitionedException {
+		Table hiveTable = getHiveTable(tablePath);
+		if (!isTablePartitioned(hiveTable)) {
+			return createCatalogTableStatistics(hiveTable.getParameters());
+		} else {
+			throw new TableNotPartitionedException(getName(), tablePath);
```

Comment:
First, shouldn't this be a `TablePartitionedException`?
Second, please correct me if I'm wrong, IIRC, if the table is partitioned, wouldn't Hive client return unknown stats?

Reply:
`hive> ANALYZE TABLE pt_products COMPUTE STATISTICS ;

FAILED: SemanticException [Error 10115]: Table is partitioned and partition specification is needed`

I test it in the hive shell and it returns an error message.